.github/workflows/check-go-task.yml

```diff
@@ -219,6 +219,55 @@ jobs:
       - name: Check whether any tidying was needed
         run: git diff --color --exit-code
 
+  check-license-headers:
+    name: check-license-headers (${{ matrix.module.path }})
+    needs: run-determination
+    if: needs.run-determination.outputs.result == 'true'
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+
+      matrix:
+        module:
+          - path: ./
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Install Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Check license headers
+        env:
+          GO_MODULE_PATH: ${{ matrix.module.path }}
+        run: |
+          # We read the template content and we modify the variadic date to be always 2000
+          LICENSE=$(cat ./license_header.tpl | sed "3s|.*|Copyright 2000 ARDUINO SA (http://www.arduino.cc/)|")
+          N_LINES=$(wc -l < ./license_header.tpl)
+          STATUS=0
+          for file in $(
+            find . \
+              -type d -name 'testdata' -prune -o \
+              -type d -name 'legacy' -prune -o \
+              -type f ! -name '*_grpc.pb.go' \
+              -type f ! -name 'json_easyjson.go' \
+              -type f ! -name 'index_easyjson.go' \
+              -regex ".*\.go$" -print
+          ); do
+          # We read only the license header, removing all the comments symbols and changing the date to be 2000. This will
+          # help us understand if the other part of the license are exactly the same as the template.
+          HEADER=$(head -n$N_LINES $file | sed -r 's| [0-9]{4} | 2000 |' | sed 's|// ||; s|^//||' )
+          if [ "$LICENSE" != "$HEADER" ]; then
+            STATUS=1
+            echo "$file"
+          fi
+          done
+          exit $STATUS
```

Comment:
I don't like this check because it just tells you that a file is wrong but didn't tell you what's wrong, and we don't have any way to re-run it locally. Also running `task go:add-license-headers` won't fix the wrong file.

Ideally, the action should just run `task go:add-license-header` and check if there are changes with `git diff`.
(more or less the same as the check for protoc files https://github.com/arduino/arduino-cli/blob/master/.github/workflows/check-protobuf-task.yml#L130).

In other words `task go:add-license-header` should write the files as they should be before pushing the PR.